pkg/arvo/sys/hoon.hoon

```diff
@@ -6124,47 +6124,52 @@
     ==
   ++  royl
     ~+
+    ;~  pose
+      (stag %rh royl-rh)
+      (stag %rq royl-rq)
+      (stag %rd royl-rd)
+      (stag %rs royl-rs)
+    ==
+  ::
+  ++  royl-rh  (cook rylh ;~(pfix ;~(plug sig sig) (cook royl-cell royl-rn)))
+  ++  royl-rq  (cook rylq ;~(pfix ;~(plug sig sig sig) (cook royl-cell royl-rn)))
+  ++  royl-rd  (cook ryld ;~(pfix sig (cook royl-cell royl-rn)))
+  ++  royl-rs  (cook ryls (cook royl-cell royl-rn))
+  ::
+  ++  royl-rn
     =+  ^=  moo
       |=  a/tape
       :-  (lent a)
       (scan a (bass 10 (plus sid:ab)))
-    =+  ^=  voy
-      %+  cook  royl-cell
-      ;~  pose
-        ;~  plug
-          (easy %d)
-          ;~  pose  (cold | hep)  (easy &)  ==
-          ;~  plug  dim:ag
-            ;~  pose
-              ;~(pfix dot (cook moo (plus (shim '0' '9'))))
-              (easy [0 0])
-            ==
-            ;~  pose
-              ;~  pfix
-                (just 'e')
-                ;~(plug ;~(pose (cold | hep) (easy &)) dim:ag)
-              ==
-              (easy [& 0])
-            ==
+    ;~  pose
+      ;~  plug
+        (easy %d)
+        ;~  pose  (cold | hep)  (easy &)  ==
+        ;~  plug  dim:ag
+          ;~  pose
+            ;~(pfix dot (cook moo (plus (shim '0' '9'))))
+            (easy [0 0])
           ==
-        ==
-        ;~  plug
-          (easy %i)
-          ;~  sfix
-            ;~  pose  (cold | hep)  (easy &)  ==
-            (jest 'inf')
+          ;~  pose
+            ;~  pfix
+              (just 'e')
+              ;~(plug ;~(pose (cold | hep) (easy &)) dim:ag)
+            ==
+            (easy [& 0])
           ==
         ==
-        ;~  plug
-          (easy %n)
-          (cold ~ (jest 'nan'))
+      ==
+      ;~  plug
```

Comment:
You'll want to put in some `::` to more clearly separate these parts of the parser (here and further down), since they're all big and multi-line. ie:

```hoon
      ==
    ::
      ;~  plug
```